src/databricks/labs/ucx/assessment/azure.py

```diff
@@ -498,6 +503,8 @@ def _map_storage(self, storage: AzureResource) -> list[StoragePermissionMapping]
                     StoragePermissionMapping(
                         prefix=f"abfss://{container.container}@{container.storage_account}.dfs.core.windows.net/",
                         client_id=role_assignment.principal.client_id,
+                        principal_id=role_assignment.principal.object_id,
+                        principal_type=role_assignment.principal.principal_type,
```

Comment:
@nfx Please review this change and see if it make sense. It's just a draft so I did not touch the tests yet. 

The idea is the `servicePrincipalType` in `directoryObjects` response would help to determine if it is a service principal ("Application") or a managed identity ("ManagedIdentity"), so my migration class can get this info from the `azure_storage_account_info.csv` and do not need to call the graph api to check again.

Another thing is I need the id/object_id/principal_id to call `directoryObjects` again to fetch the `alternativeNames` field for managed identity, it is the value I have to use to check if this managed identity is already used in UC storage credentials. (The extra `directoryObjects` call can be avoid if the `alternativeNames` is stored in `StoragePermissionMapping`, but I'm concerned it will make the `alternativeNames`  too complex.)